components/preview/src/vitepress.ts

```diff
@@ -0,0 +1,147 @@
+import { HPCCResizeElement, attribute, ChangeMap, customElement, css, display, html, ref, WebComponent } from "@hpcc-js/wc-core";
+import { HPCCCodemirrorElement } from "@hpcc-js/wc-editor";
+
+const template = html<HPCCVitepressElement>`\
+<div ${ref("_iframeDiv")}>
+</div>
+<hpcc-codemirror ${ref("_cm")}></hpcc-codemirror>`;
+
+const styles = css`
+${display("inline-block")} 
+
+:host > div {
+    flex-direction: column;
+}
+
+:host > div > div {
+    padding-bottom: 4px;
+}
+
+:host > div > hpcc-codemirror {
+    padding-top: 4px;
+}
+`;
+
+@customElement("hpcc-vitepress", { template, styles })
+export class HPCCVitepressElement extends HPCCResizeElement {
+
+    /**
+     * Border style for the preview iframe
+     *
+     * @defaultValue "1px solid #ccc"
+    */
+    @attribute preview_border = "1px solid #ccc";
+
+    /**
+     * Force full reload of iframe, on each change.
+     * 
+     * @defaultValue ""
+     */
+    @attribute head_ext = "";
+
+    /**
+     * Content to be displayed in the preview iframe
+     * 
+     * @defaultValue ""
+     */
+    @attribute content = "";
+
+    /**
+     * Split ratio between the preview and the code editor
+     * 
+     * @defaultValue 1/3
```

Comment:
Shouldn't this be `2 / 3`?

Reply:
Maybe - am planning on replacing with the layout splitpanel anyway